lib/logstash/inputs/beats.rb

```diff
@@ -257,6 +288,14 @@ def require_certificate_authorities?
     @ssl_verify_mode == "force_peer" || @ssl_verify_mode == "peer"
   end
 
+  def include_source_metadata?
+    return @enrich.include?('source_metadata')
+  end
+
+  def include_codec_metadata?
+    return @enrich.include?('codec_metadata')
+  end
+
```

Comment:
While `include_codec_metadata?` is _only_ used in `register` to effectively set `@include_codec_tag` (which is the source-of-truth for that action), and only does so when the deprecated `include_codec_tag` is not provided, its presence means that we publicly provide a method that doesn't tell the whole story.

We also should avoid reaching into `@enrich` for every single event since this is known-constant when the plugin is registered.

I've opened https://github.com/mashhurs/logstash-input-beats/pull/2 to refactor `register` to do _all_ of the work, using the alias-resolved enrich as a local variable so that we (a) don't leave traps for our future selves and (b) don't add per-event overhead.